### ISSUE TITLE
Emojione package does not work when minified for production

### ISSUE THREAD

**Issue by hashcutdev** [meteor/meteor#9647]
Meteor version: 1.6.1
OS: Ubuntu on EC2
Last time it worked: not sure, as I just upgraded emojione from the v2.2.6 branch to v3.1.2 today

I'm still getting the same problem that was listed in issue #9138 

Basically, emojione v3.1.2 works fine in development mode. It also compiles fine when either deploying or simulating deployment with meteor run --production.
However, when it's actually used in production mode, I get the following errors:

058848f859d957d1948e377e77f1ebffccba823a.js?meteor_js_resource=true:113 Uncaught SyntaxError: Invalid regular expression: /<object[^>]*>.*?</object>|<span[^>]*>.*?</span>|<(?:object|embed|svg|img|div|span|p|a)[^>]*>|(?:🏳)️?‍?(?:🌈)|(?:👁)️?‍?(?:🗨)️?|[#-9]️?⃣|(?:(?:🏴)(?:�[�-�]){1,6})|(?:�[�-�]){2}|(?:(?:�[��]))️?(?:�[�-�])?‍?(?:[⚕⚖✈]|�[�-�]|�[������])|(?:�[��]|�[�-�])(?:�[�-�])?‍?[♀♂⚕⚖✈]?️?|(?:(?:❤|�[�-��])[‍️]{0,2}){1,3}(?:❤|�[�-��])|(?:(?:❤|�[�-��])️?){2,4}|(?:�[����-���-�]|�[��]|�[�-�]|�[�-�]|�[�-�]|⛹|👯)️?(?:�[�-�])?‍?[♀♂]?️?|(?:[☝⛹✊-✍]|�[�-�]|�[�-��-��-��-�]|�[�-�])️?(?:�[�-�])|(?:[↔-↙↩-↪]️?|[#*]|[〰〽]️?|(?:�[�-�]|🆎|�[�-�])️?|Ⓜ️?|[㊗㊙]️?|(?:�[�-�]|🈚|🈯|�[�-�]|�[�-�])️?|[‼⁉]️?|[▪-▫▶◀◻-◾]️?|[©®]️?|[™ℹ]️?|🀄️?|[⬅-⬇⬛-⬜⭐⭕]️?|[⌚-⌛⌨⏏⏩-⏳⏸-⏺]️?|🃏|[⤴⤵]️?)|[✀-➿]️?|[�-�][�-�]️?|[☀-⛿]️?|[0-9]️/: Range out of order in character class
    at new RegExp (<anonymous>)
    at emojione (058848f859d957d1948e377e77f1ebffccba823a.js?meteor_js_resource=true:113)
    at 058848f859d957d1948e377e77f1ebffccba823a.js?meteor_js_resource=true:113
    at 058848f859d957d1948e377e77f1ebffccba823a.js?meteor_js_resource=true:113
    at 058848f859d957d1948e377e77f1ebffccba823a.js?meteor_js_resource=true:113

In the previous issue, this was felt to be an error with an old version of uglify-es, and that issue was closed once uglify-es was updated to the latest package. However, I'm still getting the same issue.

I've created a simple test repo at https://github.com/hashcutdev/meteor-test-emojione-minification
which reproduces the problem.

Basically, I started with a fresh install of Meteor 1.6.1, then 'meteor add emojione:emojione'. Then changed the main.html and main.js scaffolding to put in a very simple call to emojione. It works fine in development mode but gets errors in production.

I've confirmed that older versions of the emojione package work (I tested 2.2.6 which was the last of the v2 series), but not v3.0.3. Also, in my own project, I tried removing the atmosphere package and using the native NPM package with import statements and I get the same errors.

Emojione v3 was released in April '17. I personally didn't upgrade until just now, but I'm curious if people have managed to get the v3 branch working with meteor deployments? And is there a fix with the latest uglify-es to get it to work? Or is there a way to use a different minifier? (I did try completely removing the standard-minifier-js package, and type 'meteor run --production'. As expected, it ran beautifully, but the code wasn't minified, which kinda defeats the purpose :-)

**Comment by hwillson** [comment]
Sounds like it's time to bump the `uglify-es` version Meteor uses again. It's currently using 3.2.2 and `uglify-es` is now at 3.3.10. It would be great to know if version 3.3.10 fixes this issue, but we should bump the `uglify-es` version regardless. Marking as `pull-requests-encouraged` if anyone is interested. Thanks!

**Comment by abernix** [comment]
Just actually tried this a little while; I ran the repository above from a checkout with the `--production` flag and was able to reproduce it.

I updated the `uglify-es` npm dependency of `minifier-js` to `3.3.9` (the latest, [according to npm](https://npm.im/uglify-es)), and it did not fix the reproduction.  We should definitely update `uglify-es`, but it doesn't appear to be part of the scope of this issue.

**Comment by hwillson** [comment]
Thanks @abernix! 3.3.10 is on its way (it was [released earlier today](https://github.com/mishoo/UglifyJS2/releases/tag/v3.3.10) but it looks like it hasn't been published yet). Hopefully 3.3.10 fixes this issue, but if 3.3.9 didn't, then we're probably out of luck.

**Comment by klaussner** [comment]
This bug was fixed in `uglify-es` 3.3.0: https://github.com/mishoo/UglifyJS2/pull/2576 (shortly after the latest `minifier-js` update 🎯😄). The old version (3.2.2) replaced some Unicode escape sequences with the replacement character.

Snippet from the `emojione` `RegExp` with 3.2.2:
```
...\udcbb�\udd27�\ude80�...
```

And with 3.3.10:
```
...\udcbb\udcbc\udd27\udd2c\ude80\ude92...
```

**Comment by abernix** [comment]
I guess my test was flawed! (I forgot to ensure that `standard-minifier-js` rebuilt itself, since compiler packages are built into packages which consume them.).  Thanks, @klaussner!
